### ISSUE TITLE
sops_decrypt_file function doesn't work with Amazon EKS Pod Identities

### ISSUE THREAD

**Issue by jeremiahsnapp** [gruntwork-io/terragrunt#3249]
## Describe the bug

[Amazon EKS Pod Identities](https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html) was added to EKS clusters late last year. It provides a more native way to manage credentials for applications running in EKS clusters. The application must use a new enough version of AWS SDK to successfully retrieve credentials from the new Pod Identities API.

Terragrunt's [sops_decrypt_file](https://terragrunt.gruntwork.io/docs/reference/built-in-functions/#sops_decrypt_file) function doesn't work with Pod Identities because Terragrunt depends on [SOPS version 3.8.1](https://github.com/getsops/sops/blob/v3.8.1/go.mod) which depends on an older version of AWS SDK Go v2 that doesn't support Pod Identities.

reference: [Minimum AWS SDK version requirements for Pod Identities](https://docs.aws.amazon.com/eks/latest/userguide/pod-id-minimum-sdk.html)

[SOPS version 3.9.0](https://github.com/getsops/sops/blob/v3.9.0/go.mod) was released a couple weeks ago and I verified that it works with Pod Identities. Upgrading Terragrunt's dependency on SOPS to version 3.9.0 will resolve this issue.

## Steps To Reproduce

Create an EKS pod that uses EKS Pod Identities to retrieve AWS credentials. Run Terragrunt in the pod with a `terragrunt.hcl` file that uses the `sops_decrypt_file` function to decrypt a sops file.

## Expected behavior

Terragrunt's `sops_decrypt_file` function should work when running Terragrunt in an EKS pod that uses Pod Identities.

## Versions

- Terragrunt version: 0.59.6
- Terraform version: 1.9.0
- [alpine/terragrunt:1.9.0 container](https://hub.docker.com/r/alpine/terragrunt)

**Comment by denis256** [comment]
Looks like it is duplicate of https://github.com/gruntwork-io/terragrunt/issues/3246

**Comment by jeremiahsnapp** [comment]
Yes, you're correct. Sorry I didn't see that when I checked a couple days ago and I didn't think to check again today. I'll close this one.
